### DISCUSSION TITLE
Multiple colored lines for relationship visualization

### DISCUSSION THREAD

**Discussion by itsalexschwartz** [natefrisch01/Graph-Link-Types#24] [Ideas]
Hey Nate! Thanks for kicking off this project :)

I have a specific request / use case of being able to visualize relationships where multiple people have multiple types of connections to one another.. 

The idea would be to draw multiple colored lines that lie next to one another to represent the various types of links, with multiple link types able to be defined.

Here's the note:
<img width="121" alt="Screenshot 2024-03-08 at 2 39 34 PM" src="https://github.com/natefrisch01/Graph-Link-Types/assets/48920346/6ca6dee5-8250-4807-9e88-375f42929212">

Here's my intended outcome (mockup):
![1](https://github.com/natefrisch01/Graph-Link-Types/assets/48920346/e1b7f4ee-3d08-4860-a3c8-1e77e3edb312)

Thoughts? The text labels end up actually being mostly unnecessary if one can display a legend of line colors in the corner of the graph. And of course being able to define multiple links in-line like Partner&Friend::Ali would be nice for ease of defining but not necessary to get the basics running.

Buying you a coffee for your time to think on this :P

**Comment by natefrisch01** [comment]
Hey @itsalexschwartz thanks for the coffee! This is definitely something we can implement - the hardest thing will be making a UI for choosing colors associated with each tag. Perhaps a default color with for each tag could be generated, and then customized in the UI and colors/tags could be turn on/off as well.

**Reply by itsalexschwartz** [reply]
Glad to hear it doesn't sound outlandish to pull off. 

And yeah a default color per tag sounds great. Customizing the colors for each tag seems like a wonderful thing to be able to define but not needed for min-spec so long as colors stayed consistent and were visually distinguishable.

**Reply by natefrisch01** [reply]
Totally true. I'll plan on trying to add this min-spec in a near future release.

**Reply by itsalexschwartz** [reply]
juggl allows for something close with their Link Types and CSS styling (seen here)
<img width="258" alt="Screenshot 2024-03-08 at 3 56 00 PM" src="https://github.com/natefrisch01/Graph-Link-Types/assets/48920346/52cc7c90-c258-4a25-94f9-eacba51744da">

The issue is they dont support multiple different links between two nodes. You mention the hardest thing would be to make a UI for choosing colors -- my uninformed guess is that the difficulty lies in rendering two or more link lines next to one another  in the graph. Thoughts?

**Reply by natefrisch01** [reply]
I don't anticipate it being too much of an issue. With multiple links, there is already slightly different logic handling how the text is drawn, so fingers crossed I can just use the same logic when drawing the links. (forgive the lack of error handling in the else, lol)

From:
`linkManager.createTextForLink`
```typescript
        if (pairStatus === LinkPair.None) {

        } else if (pairStatus === LinkPair.First) {
            linkString = linkString + "\n\n";
        } else if (pairStatus === LinkPair.Second) {
            linkString = "\n\n" + linkString;
        } else {

        }
```

**Comment by natefrisch01** [comment]
I have a rough version up on the latest release. You have to enable it in the settings tab, under Community Plugins: Graph Link Types. I have found a lot of ways to break it, but with a little refinement I think it will be quite useful. Let me know what you all think!
![image](https://github.com/natefrisch01/Graph-Link-Types/assets/44580969/a4a7d6f7-3442-47ad-ba19-7d70b8682bd1)

**Reply by natefrisch01** [reply]
![image](https://github.com/natefrisch01/Graph-Link-Types/assets/44580969/40165858-77f8-42da-985e-c2dc30e67df9)
![2024-03-1811-06-26-ezgif com-video-to-gif-converter](https://github.com/natefrisch01/Graph-Link-Types/assets/44580969/5bdda828-ecc2-4949-a5d0-0a7be22d877b)

**Reply by itsalexschwartz** [reply]
oh my!! This is incredible! I'm about to go test it out :)  Great work Nate!

**Reply by itsalexschwartz** [reply]
This is working perfectly for my needs. A++. And the legend works beautifully!

Based on what I'm using this for, the next thing I'd need is the ability for two nodes to be able to define more than 2 tagged links between one another so I can draw 3+ lines between two nodes.
